File: 2-Patron/php/MascotaGrande.php
<?php
class MascotaGrande
{
  private $montura;
  private $armadura;
  private $ultimate;

  function __construct($nombre, $defensa, $ataque, $nivel)
  {
    parent::__construct($nombre, $defensa, $ataque, $nivel);
    $this->montura = null;
    $this->armadura = null;
    $this->ultimate = null;
  }

  function getMontura()
  {
    return $this->montura;
  }

  function setMontura($nMontura)
  {
    $this->montura = $nMontura;
  }

  function getArmadura()
  {
    return $this->armadura;
  }

  function setArmadura($nArmadura)
  {
    $this->armadura = $nArmadura;
  }

  function getUltimate()
  {
    return $this->ultimate;
  }

  function setUltimate($nUltimate)
  {
    $this->ultimate = $nUltimate;
  }

  function usarUltimate()
  {
    return $this->ultimate;
  }

  function atacar()
  {
    return $this->ataque * ($this->nivel * 0.5);
  }

  function decSalud($decremento)
  {
    $this->salud -= $decremento / $this->defensa;
  }

  function incNivel()
  {
    $this->nivel += 1;
  }

  function verStats()
  {
    return  parent::verStats() .
            "Montura: " . $this->montura .
            "Armadura: " . $this->armadura .
            "Ultimate: " . $this->ultimate;
  }
}
?>

File: 2-Patron/Java/MascotaGrande.java
package abstractmasterwhole;
public class MascotaGrande extends Criatura{
    private Montura montura = null;
    private Armadura armadura = null;
    private Ultimate ultimate = null;
    String nombre;
    int defensa;
    int ataque;
    int nivel;
    public MascotaGrande(String nombre, int defensa, int ataque, int nivel) {
        super(nombre, defensa, ataque, nivel);
                
   
    }
    public Montura getMontura(){
        return this.montura;
    }
    public void setMontura(Montura montura) {
        this.montura = montura;
    }
    public Armadura getArmadura(){
        return this.armadura;
    }

    public void setArmadura(Armadura armadura) {
        this.armadura = armadura;
    }
    public Ultimate getUltimate() {
        return this.ultimate;
    }

    public void setUltimate(Ultimate ultimate) {
        this.ultimate = ultimate;
    }
    public Ultimate usarUltimate(){
        return this.ultimate;
    }
    public float Atacar(){
        
        return (ataque*(nivel*0.5));
    }
    
    @Override
    public int decSalud(int dec) {
        return this.salud -= dec/this.defensa;
    }
    
    @Override
    public int incNivel(){
        nivel += 1;
        return nivel;
    }
    
    
    @Override
    public String verStats(){
       
        return (super.verStats()+ "\n Montura: "+ this.montura+ "\nArmadura: "+ this.armadura+ "\nUltimate: "+ this.ultimate);
    }
    
    
  
}
